icechunk-python/tests/test_stateful_repo_ops.py

```diff
@@ -362,6 +388,63 @@ def commit(self, message: str) -> str:
         self.commit_times.append(snapinfo.written_at)
         return commit_id
 
+    @rule(message=st.text(max_size=MAX_TEXT_SIZE), target=commits)
+    @precondition(lambda self: self.model.changes_made)
+    def amend_commit(self, message: str) -> str:
+        """Amend the last commit.
+
+        Amend requires spec_version >= 2. For spec_version=1, it raises an error.
+        """
+        branch = self.session.branch
+        assert branch is not None
+        note(
+            f"Amending commit on branch {branch!r} (spec_version={self.model.spec_version})"
+        )
+
+        # Amend is only supported on spec_version >= 2
+        if self.model.spec_version == 1:
```

Comment:
prefer just making a precondition to avoid this; and use a unit test for this error (its trivial)